opentelemetry-sdk/src/opentelemetry/sdk/context/threadlocal_context.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+from opentelemetry.context import Context, RuntimeContext
+
+_CONTEXT_KEY = "current_context"
+
+
+class ThreadLocalRuntimeContext(RuntimeContext):
+    """An implementation of the RuntimeContext interface
+    which uses thread-local storage under the hood. This
+    implementation is available for usage with Python 3.4.
+    """
+
+    def __init__(self) -> None:
+        self._current_context = threading.local()
+
+    def set_current(self, context: Context) -> None:
+        """See `opentelemetry.context.RuntimeContext.set_current`."""
+        setattr(self._current_context, _CONTEXT_KEY, context)
+
+    def get_current(self) -> Context:
+        """See `opentelemetry.context.RuntimeContext.get_current`."""
+        try:
```

Comment:
I think this `try`/`except` is no longer necessary if `self._current_context._CONTEXT_KEY` is initialized to `Context`.

Reply:
I think we want to keep the try/except here. In the case we try to get_current on a new thread, the previously initialized value will not be in thread local.

Reply:
:thinking: but what we are doing here is looking for `self._current_context` to have the `_CONTEXT_KEY` attribute, and that should always be there if we [add](https://github.com/open-telemetry/opentelemetry-python/pull/395/files/f8032c1c355f9664cef108d406ef1d7a6b0418ad#r377938431) it in `__init__`, or am I missing something here? :man_shrugging:

Reply:
Right, but `_current_context` is a thread local object:

`E       AttributeError: '_thread._local' object has no attribute 'current_context'`

Reply:
ah, right, got it :+1: